src/wallet/asyncrpcoperation_sendmany.cpp

```diff
@@ -803,6 +803,10 @@ bool AsyncRPCOperation_sendmany::find_utxos(bool fAcceptCoinbase=false) {
     pwalletMain->AvailableCoins(vecOutputs, false, NULL, true, fAcceptCoinbase);
 
     BOOST_FOREACH(const COutput& out, vecOutputs) {
+        if (!out.fSpendable) {
```

Comment:
I'm wondering if in terms of efficiency, we should do these checks of `fSpendable` and `nDepth` inside an altered version of `AvailableCoins`.  Cause right now we are scanning all the outputs twice.

Reply:
But I'm fine with doing this optimization later, if we want to fix it like this now.